### ISSUE TITLE
include vaccinations and virus variants in get_derivatives of vaccination model

### ISSUE THREAD

**Issue by dabele** [DLR-SC/memilio#235]
Currently, vaccinations and virus variants are applied in `secirv::Simulation::apply` by modifying parameters or directly setting compartments. I think this is not ideal, as they could be included more elegantly in `secirv::Model::get_derivatives`. 

In general we should limit the places in the code where compartments or parameters are modifed, otherwise the model becomes very hard to follow.

E.g., daily first vaccinations are the flow from compartment `Susceptible` to `SusceptiblePartiallyImmune`.

**Comment by mknaranja** [comment]
A base for this discussion could be what we had implemented in the parameter `DynamicInfectionFromContact` here:

https://github.com/DLR-SC/memilio/pull/233/files/cda84d1dd6db5bdc8a5b67354dcb65f443bc52d1#diff-ec5f30db7521beeba42314e5ef7c1e097edea1667dc21478f55ab0e6885a6cbbR152-R166

**Comment by jubicker** [comment]
outdated
